pulsar-client-cpp/lib/Murmur3_32.cc

```diff
@@ -0,0 +1,108 @@
+/**
```

Comment:
@Licht-T - As per you comment in https://github.com/apache/incubator-pulsar/issues/1017
> boost::hash uses MurmurHash3

Then why are we reimplementing it and changing the code on C++ side

Reply:
@jai1 There is various implementations of MurmurHash3. `boost::hash` is much different from the Google Guava's one which is based on the original one.

Reply:
Is the actual result different between Guava and boost versions?

Reply:
@merlimat Yes, and the result by my implementation is same as Google Guava's.
Google Guava: https://ideone.com/Jq4stk
`boost::hash`: https://ideone.com/W4jEBT

Reply:
Nice! Thanks for checking